### ISSUE TITLE
Command line -s parameters ignored

### ISSUE THREAD

**Issue by jleroy** [conan-io/conan#845]
Using Conan `0.18.0`, the `-s` parameters seems to be ignored.

Here's the settings presents in my `conan.conf` file:
```
[settings_defaults]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=6.3
os=Linux
```

I want to force the use of GCC 6.2 instead of 6.3:
```
$ conan install .. -s compiler=gcc -s compiler.libcxx=libstdc++ -s compiler.version=6.2 --build
ERROR: '6.3' is not a valid 'settings.compiler.version' value.
Possible values are ['4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5.1', '5.2', '5.3', '5.4', '6.1', '6.2']
Check your settings ~/.conan/settings.yml or read the docs FAQ
``` 

As you can see, the `compiler.version=6.2` setting is not taken into account. However, if I use the `CONAN_ENV_COMPILER_VERSION` environment variable, it works:
```
export CONAN_ENV_COMPILER_VERSION=6.2
$ conan install ..
web-client-tv/2.10.8-fd540f9@plex/public: Not found, looking in remotes...
[...]
```

Maybe this is relates to the recent changes introduced by #802?

**Comment by memsharded** [comment]
Maybe it didn't run the migrations properly. Please try to remove ``conan.conf`` and launch again, or add ``6.3`` to your ``<userhome>/.conan/settings.yml`` file.

The issue is that even if you are defining ``6.2`` in the command line, the default ``6.3`` is still validated against the ``sèttings.yml`` before being overwritten. TL;DR: the `conan.conf`` defaults must exist in ``settings.yml``.

Please tell me if that solves the issue. Thanks!

**Comment by jleroy** [comment]
Conan `0.18.0`does not support GCC 6.3. 
But yes, it works if a add `6.3` to my `~/.conan/settings.yml` file manually.

So this mean that using the `CONAN_ENV_COMPILER_VERSION` environment variable bypass the `conan.conf` validation, but the `-s compiler.version=6.2` command line parameter does not?

**Comment by lasote** [comment]
I setted `CONAN_ENV_COMPILER_VERSION` to `6.8` to test it and the result is:

```
ERROR: '6.8' is not a valid 'settings.compiler.version' value.
Possible values are ['4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5.1', '5.2', '5.3', '5.4', '6.1', '6.2']
Check your settings ~/.conan/settings.yml or read the docs FAQ
```

So it seems that yes, it also check it.

**Comment by jleroy** [comment]
Yes, but the value of `compiler.version` setting in `conan.conf` is not checked.

**Comment by lasote** [comment]
Solved in PR #885 pending to merge

**Comment by memsharded** [comment]
Released in conan 0.19
